File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<string> babbling) {
    int answer = 0;
    vector<string> can_pronounce = {"aya", "ye", "woo", "ma"};
    
    for(int i=0;i<babbling.size();i++){
        bool is_valid = true;
        for(int j=0;j<babbling[i].size();j++){
            if(babbling[i].substr(j,3) == "aya") j += 2;
            else if(babbling[i].substr(j,2) == "ye") j += 1;
            else if(babbling[i].substr(j,3) == "woo") j += 2;
            else if(babbling[i].substr(j,2) == "ma") j += 1; 
            else{
                is_valid = false;
                break;
            } 
        }
        
        if (is_valid){
            answer ++;
        }
    }
    
    return answer;
}

File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.cs
using System;

public class Solution {
    public int solution(string[] babbling) {
        int answer = 0;
            
        for (int i = 0; i < babbling.Length; i++) {
            string bab = babbling[i];
            bab = bab.Replace("aya",".");
            bab = bab.Replace("ye",".");
            bab = bab.Replace("woo",".");
            bab = bab.Replace("ma",".");
            
            Console.WriteLine(bab);
            
            bab = bab.Replace(".","");

            if(bab == "")
            {
                answer++;
            }   
        }

        
        return answer;
    }
}
